tapir/coop/tests/factories.py

```diff
@@ -81,11 +80,11 @@ class MembershipResignationFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = MembershipResignation
 
-    share_owner = factory.Subfactory(ShareOwnerFactory)
-    cancellation_reason = factory.Faker("Resigned for testing reason.")
+    share_owner = factory.SubFactory(ShareOwnerFactory)
+    cancellation_reason = factory.Faker("sentence")
     transferring_shares_to = factory.SubFactory(ShareOwnerFactory)
-    resignation_type = factory.Faker(
-        random.random_choice(MembershipResignation.resignation_type)
+    resignation_type = factory.random.reseed_random(
+        MembershipResignation.ResignationType
```

Comment:
I'm not sure what you're trying to do here?
I think you need something like [here](https://stackoverflow.com/a/49239188) with `random_element`